app/src/main/java/org/traccar/client/MainActivity.java

```diff
@@ -307,4 +289,38 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         }
     }
 
+    private boolean validateServerURL(String userUrl) {
+        if (userUrl == null || userUrl.trim().length() == 0) {
+            EditTextPreference preference = (EditTextPreference) findPreference(KEY_URL);
+            preference.setText(getString(R.string.settings_url_default_value));
+            return false;
+        }
+        int port = Uri.parse(userUrl).getPort();
+        if (URLUtil.isValidUrl(userUrl) && (port >= -1 && port <= 65535)
```

Comment:
`getPort()` returns `-1` if no port was set in which case default port `5055` will be used as per your former request